homeassistant/components/recorder/statistics.py

```diff
@@ -846,7 +930,9 @@ def _flatten_list_statistic_ids_metadata_result(
         {
             "statistic_id": _id,
             "display_unit_of_measurement": info["display_unit_of_measurement"],
-            "has_mean": info["has_mean"],
+            "has_mean": info["mean_type"]
+            == StatisticMeanType.ARIMETHIC,  # Can be removed with 2026.4
+            "mean_type": info["mean_type"],
```

Comment:
Why are we backwards compatible here, but not if `async_list_statistic_ids` or `list_statistic_ids` are called without statistic ids?

Reply:
This method is called in the both mentioned functions as last statement. Adding it here will add backwards compatibility for both